sdk/rust/examples/contract_deploy.rs

```diff
@@ -0,0 +1,1596 @@
+    )?;
+
+    match command {
+        Command::Empty => {
+            println!("Params:\nblockfrost: {:?}\nsubmit: {}", config, submit);
+            Ok(())
+        }
+
+        Command::Lock(params) => handle_lock(params, config, submit).await,
+        Command::LockNft(params) => handle_lock_nft(params, config, submit).await,
+        Command::Unlock(params) => handle_unlock(params, config, submit).await,
+        Command::Claim(params) => handle_claim(params, config, submit).await,
+    }
+}
+
+async fn handle_lock(
+    config: ConfigPath,
+    blockfrost: BlockfrostConfiguration,
+    submit: bool,
+) -> anyhow::Result<()> {
+    let config: LockConfiguration = serde_yaml::from_reader(
+        File::open(config.config).context("failed to open the delegate rewards config file")?,
+    )?;
+
+    let network = blockfrost.get_network();
+    println!("Network:\t{network}");
+
+    let payment_address = get_payment_creds(
+        network,
+        config.payment_configuration.payment_vkey,
+        config.payment_configuration.payment_bech32,
+    )?;
+    println!("Address:\t{}", payment_address.to_bech32(None).unwrap());
+
+    let mut builder = create_tx_builder();
+
+    let blockfrost = Blockfrost::new(blockfrost)?;
+    let (_, inputs) = fetch_inputs(config.inputs.clone(), &blockfrost).await?;
+
+    for (pointer, input) in inputs {
+        builder
+            .add_input(
+                SingleInputBuilder::new(
+                    TransactionInput::new(pointer.hash, pointer.index),
+                    TransactionOutput::new(payment_address.clone(), input, None, None),
+                )
+                .payment_key()
+                .unwrap(),
+            )
+            .unwrap();
+    }
+
+    // collateral
+
+    if config.receipt_nft.is_some() {
+        let (_, collateral) = fetch_inputs(vec![config.collateral], &blockfrost).await?;
+
+        for (pointer, input) in collateral {
+            builder
+                .add_collateral(
+                    SingleInputBuilder::new(
+                        TransactionInput::new(pointer.hash, pointer.index),
+                        TransactionOutput::new(payment_address.clone(), input, None, None),
+                    )
+                    .payment_key()
+                    .unwrap(),
+                )
+                .unwrap();
+        }
+    }
+
+    let now = std::time::SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    let validity = ttl_by_posix(now - 100);
+    let ttl = ttl_by_posix(now + 100);
+
+    let script = PlutusV2Script::from_bytes(hex::decode(config.lock_on).unwrap()).unwrap();
+    let contract = PlutusScript::PlutusV2(script);
+
+    let lock_on =
+        EnterpriseAddress::new(network, Credential::new_script(contract.hash())).to_address();
+    println!("Lock on contract: {}", lock_on.to_bech32(None).unwrap());
+
+    // let receipt_asset_name = if let Some(nft) = config.receipt_nft.clone() {
```

Comment:
Do we need this large block of commented code?

Reply:
if we are ever going to use receipt then yes, it will help